src/pykeen/ablation/ablation.py

```diff
@@ -34,88 +33,164 @@ def ablation_pipeline(
     losses: Union[str, List[str]],
     optimizers: Union[str, List[str]],
     training_loops: Union[str, List[str]],
-    *,
     create_inverse_triples: Union[bool, List[bool]] = False,
     regularizers: Union[None, str, List[str]] = None,
+    negative_sampler: Union[str, None] = None,
+    evaluator: Optional[str] = None,
+    stopper: Optional[str] = 'NopStopper',
     model_to_model_kwargs: Optional[Mapping2D] = None,
     model_to_model_kwargs_ranges: Optional[Mapping2D] = None,
-    model_to_trainer_to_training_kwargs: Optional[Mapping3D] = None,
-    model_to_trainer_to_training_kwargs_ranges: Optional[Mapping3D] = None,
-    ablation_config: Optional[Mapping3D] = None,
-    evaluator: Optional[str] = None,
-    optuna_config: Optional[Mapping[str, Any]] = None,
+    model_to_loss_to_loss_kwargs: Optional[Mapping3D] = None,
+    model_to_loss_to_loss_kwargs_ranges: Optional[Mapping3D] = None,
+    model_to_optimizer_to_optimizer_kwargs: Optional[Mapping3D] = None,
+    model_to_optimizer_to_optimizer_kwargs_ranges: Optional[Mapping3D] = None,
+    model_to_negative_sampler_to_negative_sampler_kwargs: Optional[Mapping3D] = None,
+    model_to_negative_sampler_to_negative_sampler_kwargs_ranges: Optional[Mapping3D] = None,
+    model_to_training_loop_to_training_loop_kwargs: Optional[Mapping3D] = None,
+    model_to_training_loop_to_training_kwargs: Optional[Mapping3D] = None,
+    model_to_training_loop_to_training_kwargs_ranges: Optional[Mapping3D] = None,
+    model_to_regularizer_to_regularizer_kwargs: Optional[Mapping3D] = None,
+    model_to_regularizer_to_regularizer_kwargs_ranges: Optional[Mapping3D] = None,
     evaluator_kwargs: Optional[Mapping[str, Any]] = None,
     evaluation_kwargs: Optional[Mapping[str, Any]] = None,
-    directory: Optional[str] = None,
-    dry_run: bool = False,
-    best_replicates: Optional[int] = None,
+    stopper_kwargs: Optional[Mapping[str, Any]] = None,
+    n_trials: Optional[int] = 5,
+    timeout: Optional[int] = 3600,
+    metric: Optional[str] = 'hits@10',
+    direction: Optional[str] = 'maximize',
+    sampler: Optional[str] = 'random',
+    pruner: Optional[str] = 'nop',
+    metadata: Optional[Mapping] = None,
+    directory: str = None,
```

Comment:
before the type annotation was Optional[str]. This type annotation doesn't make sense, and there isn't any code for handling the none case anymore. I'm going to turn on MyPy for this file to catch stuff like this.

Reply:
Is there an argument against making this argument mandatory (liek done in d17ea2ba010a222581ded1256264687d8183b4c1)?

Reply:
nope, but you could always use the `pykeen.constants.PYKEEN_EXPERIMENTS` tool to make a default directory if none is given

Reply:
Then I would opt for asking the user to define the output directory since, in larger ablation studies, the trained models may already occupy a decent amount of memory, and they should not be saved in an unexpected place :)

Reply:
fine with me! if they are defining it explicitly, then maybe you should change the part that makes a subdirectory with the date/time to be optional

Reply:
Done in a310e5c6f4ecdb986dbdfca5ae21ac3e94ce0cf0.